### ISSUE TITLE
Broker ping or latency

### ISSUE THREAD

**Issue by afzl-wtu** [shamblett/mqtt_client#556]
I can have a callback function when pong response arrives. But I do not get what is in response. Is there any way to calculate or get Network latency of client and broker?

**Comment by shamblett** [comment]
There is nothing in the pong response message, its empty, so there's nothing in it to use to calculate latency etc. So no , there is no way at the moment to calculate network latency between ping request and pong response.

I believe this is useful to users so I could add a ping callback, this will be called when a ping is sent to the broker, one way to calculate latency is to  set a timestamp and compare it to the reception time of the pong response or maybe the client can maintain a running value of latency itself that users can then just read.

Any thoughts on what you would prefer here?

**Comment by afzl-wtu** [comment]
@shamblett
Yes, if we get Ping call back and then pong call back we can calculate ping or latency. 
It would be even better instead of these callbacks just to give calculated ping stream. Because I do not think if any person uses pong call backm

**Comment by shamblett** [comment]
The pong callback is used by many users to detect if the broker goes away somehow, if a pong callback is not called for a certain amount of time then the user can take whatever steps needed for their particular use case. There is no equivalent usage for a ping callback however, hence it was never supplied.

To be flexible I think I'll add a ping callback for users who wish to calculate latency themselves and also add a measure of the last ping/pong cycle latency and a running average for a connect period, both in milliseconds.

This should suffice for most(or all) users.

**Comment by shamblett** [comment]
Updates complete, package re released at version 10.5.0.

Please see the mqtt_server_client.dart example for usage examples and refer to the updated package API documentation.
